ee/src/main/java/org/jboss/as/ee/component/deployers/ResourceReferenceProcessor.java

```diff
@@ -272,6 +272,8 @@ public ManagedReference getReference() {
             final BindingConfiguration bindingConfiguration;
             if (!isEmpty(lookup)) {
                 bindingConfiguration = new BindingConfiguration(name, new LookupInjectionSource(lookup));
+            } else if (isEmpty(value)) {
+                throw EeLogger.ROOT_LOGGER.cannotLoad(value);
```

Comment:
I think it is detected because of the && condition.:L254=>_both_ fields with value; L256 _both_ fields null.
I'll try the assert and come back - interesting question in general. I can document this with "Won't fix" - any solution should help with error prevention or ease analysis. Some of my managers would do it to decrease the total number of findings (regardless of the sense) - I don't like "Key Performance Indicator" work :-)

Reply:
@bstansberry `assert !isEmpty(lookup) || value != null;` is not recognized by sonarcloud. I'll remove ResourceReferenceProcessor from this PR and mark it as won't fix.